### PR TITLE
feat: implement a new mode `try` in the config manipulation commands

### PR DESCRIPTION
**Pull request by Unix4ever**
Fixes: #4591 

The new mode allows changing the config for a period of time, which
allows trying the configuration and automatically rolling it back in case
if it doesn't work for example.

The mode can only be used with changes that can be applied without a
reboot.

When changed it doesn't write the configuration to disk, only changes it
in memory.
`--timeout` parameter can be used to customize the rollback delay.
The default timeout is 1 minute.

Any consequent configuration change will abort try mode and the last
applied configuration will be used.

Signed-off-by: Artem Chernyshev <artem.chernyshev@talos-systems.com>

---
This change is [Reviewable](https://reviewable.io/reviews/siderolabs/talos/5400)

### CODE REVIEW

File: internal/app/machined/internal/server/v1alpha1/v1alpha1_server.go [hunk 1]
```diff
@@ -244,6 +268,10 @@ Config diff:
 		return nil, fmt.Errorf("incorrect mode '%s' specified for the apply config call", in.Mode.String())
 	}
 
+	if in.Mode != machine.ApplyConfigurationRequest_TRY {
```
[the remark this answers is not in the document]
Comment (Unix4ever) [reply internal/app/machined/internal/server/v1alpha1/v1alpha1_server.go]: makes sense, updated

**Review by frezbo — APPROVED** [review]

**Review by smira — APPROVED** [review]

**Pull request merged**